lib/std/c/test.zig

```diff
@@ -0,0 +1,21 @@
+const std = @import("std");
+const c = std.c;
+const os = std.os;
+const testing = std.testing;
+
+test "getaddrinfo" {
+    const port = 9999;
+    const port_c = try std.fmt.allocPrint(testing.allocator, "{}\x00", .{port});
```

Comment:
You might as well hard-code `"9999"`

Reply:
Hey, would love for you to check to make sure I did this right. I understand the previous implementation to be doing two things, specifically allocating the strong from an int, and appending the null terminator. It's not entirely clear to me if I simply create a string whether or not it will imply null termination or not, I've biased towards not and included the null byte.

Reply:
You don't need the null-terminator, since strings are normally null terminated and the type argument ``noalias service: [*:0]const u8,`` makes sure of that.